wordcount/count/FileOpenHandleAsciiCheatBenchmark.cs

```diff
@@ -0,0 +1,72 @@
+using System.Buffers;
+using System.Diagnostics;
+using BenchmarkData;
+
+namespace FileOpenHandleAsciiCheatBenchmark;
+
+public static class FileOpenHandleAsciiCheatBenchmark
+{
+    public static Count Count(string path)
+    {
+        const byte NEWLINE = (byte)'\n';
+        const byte SPACE = (byte)' ';
+        ReadOnlySpan<byte> searchValues = [SPACE, NEWLINE];
+
+        int wordCount = 0, lineCount = 0, byteCount = 0;
+        bool wasSpace = true;
+
+        byte[] buffer = ArrayPool<byte>.Shared.Rent(BenchmarkValues.Size);
+        using var handle = File.OpenHandle(path, FileMode.Open, FileAccess.Read, FileShare.Read, FileOptions.SequentialScan);
+
+        // Read content in chunks, in buffer, at count lenght, starting at byteCount
+        int count = 0;
+        while ((count = RandomAccess.Read(handle, buffer, byteCount)) > 0)
+        {
+            byteCount += count;
+            Span<byte> bytes = buffer.AsSpan(0, count);
+            
+            while (bytes.Length > 0)
+            {
+                // what's this character?
+                if (bytes[0] <= SPACE)
+                {
+                    if (bytes[0] is NEWLINE)
+                    {
+                        lineCount++;
+                    }
+
+                    wasSpace = true;
+                    bytes = bytes.Slice(1);
+                    continue;
+                }
+                else if (wasSpace)
+                {
+                    wordCount++;
+                }
+
+                // Look ahead for next space or newline
+                // this logic assumes that preceding char was non-whitespace
+                int index = bytes.IndexOfAny(searchValues);
+
+                if (index > -1)
+                {
+                    if (bytes[index] is NEWLINE)
+                    {
+                        lineCount++;
+                    }
+
+                    wasSpace = true;
+                    bytes = bytes.Slice(index + 1);
+                }
+                else
+                {
+                    wasSpace = false;
+                    bytes = [];
+                }
+            }
+        }
+
+        ArrayPool<byte>.Shared.Return(buffer);
+        return new(lineCount, wordCount, byteCount, path);
```

Comment:
Bug: A file containing just the 5-byte content `hello` will show up as wordCount = 0, lineCount = 0, bytecount = 5.

Reply:
That's not what I see when I run the code. I made some tests (which I should have done earlier).

```bash
[rich@kelowna count]$ wc ../tests/*
 4  0 14 ../tests/no-words-with-spaces-newlines.txt
 1  1  5 ../tests/word-no-space.txt
 3  1  9 ../tests/word-with-newlines.txt
 1  1  7 ../tests/word-with-spaces.txt
 9  3 35 total
[rich@kelowna count]$ dotnet run ../tests/
FileOpenHandleRuneBenchmark
        4       0      14 ../tests/no-words-with-spaces-newlines.txt
        1       1       5 ../tests/word-no-space.txt
        3       1       9 ../tests/word-with-newlines.txt
        1       1       7 ../tests/word-with-spaces.txt
        9       3      35 total
```